src/Package/Impl/DataInspect/VisualGrid/VisualGrid.cs

```diff
@@ -65,36 +64,43 @@ internal sealed class VisualGrid : FrameworkElement {
             }
         }
 
-        public static readonly DependencyProperty FontSizeProperty =
-                TextElement.FontSizeProperty.AddOwner(
-                        typeof(VisualGrid));
+        public static readonly DependencyProperty FontSizeProperty = TextElement.FontSizeProperty.AddOwner(typeof(VisualGrid));
 
         [TypeConverter(typeof(FontSizeConverter))]
         public double FontSize {
             get { return (double)GetValue(FontSizeProperty); }
-            set { SetValue(FontSizeProperty, value); }
+            set {
+                _typeFace = null;
+                SetValue(FontSizeProperty, value);
+            }
+        }
+
+        public static readonly DependencyProperty FontWeightProperty = TextElement.FontWeightProperty.AddOwner(typeof(VisualGrid));
+
+        [TypeConverter(typeof(FontWeightConverter))]
+        public FontWeight FontWeight {
+            get { return (FontWeight)GetValue(FontWeightProperty); }
+            set {
+                _typeFace = null;
+                SetValue(FontWeightProperty, value);
+            }
         }
 
         private Typeface _typeFace;
         private Typeface Typeface {
             get {
                 if (_typeFace == null) {
-                    _typeFace = ChooseTypeface();
+                    if (FontFamily != null && FontSize > 0) {
```

Comment:
Normally it is not exposed AFACT. Controls have FontSize/Weight/Family/Stretch, never seen Typeface though. I guess this is b/s 4 former ones are defining it and not a single property.